crates/js-sys/src/lib.rs

```diff
@@ -2050,6 +2050,152 @@ extern "C" {
     pub fn set_prototype_of(target: &Object, prototype: &JsValue) -> bool;
 }
 
+// RegExp
+#[wasm_bindgen]
+extern {
+    #[derive(Clone, Debug)]
+    pub type RegExp;
+
+    /// The exec() method executes a search for a match in a specified
+    /// string. Returns a result array, or null.
+    ///
+    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec
+    #[wasm_bindgen(method)]
+    pub fn exec(this: &RegExp, text: &str) -> Option<Array>;
```

Comment:
MDN states that on a successful match the returned value is an array. But in the example later on one can see that there are also properties `index` and `input` added to this array. I put Array here for now as I don't know how to add those properties. How can that be done?

Reply:
I think it's fine to return an `Array` for now, but the alternative is you could return a custom type here with `input` and `index` properties defined. For now though it's likely ok to rely on users to do that locally if they need to access the values.